### PR TITLE
DPL: fix race condition with the delivery of region events

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
@davidrohr this is my understanding at the end of our discussion today.

**Comment by davidrohr** [comment]
Looks good.
Just a cosmetic comment: I don't understand the comment in the code about breaking out of the loop.

File: Framework/Core/src/DataProcessingDevice.cxx [hunk 1]
```diff
@@ -991,14 +994,21 @@ void DataProcessingDevice::InitTask()
...
   /// on the main thread.
   /// * Wait for enough callbacks to be delivered before moving to START
-  while (deviceContext.expectedRegionCallbacks > 0 && uv_run(state.loop, UV_RUN_ONCE)) {
+  while (hasPendingEvents(deviceContext)) {
+    // Wait for the callback to signal its done, so that we do not busy wait.
+    uv_run(state.loop, UV_RUN_ONCE);
     // Handle callbacks if any
     {
       std::lock_guard<std::mutex> lock(mRegionInfoMutex);
+      // If there is no pending event, we can break the loop.
```
Comment (ktf) [inline Framework/Core/src/DataProcessingDevice.cxx]:
```suggestion
```

remove leftover

**Comment by ktf** [comment]
Just a leftover from a previous iteration which I did not spot on rebase.

**Pull request merged**